### PR TITLE
Added segment_sum function to math.py

### LINKED ISSUE #21903: segment_mean

### PR DESCRIPTION
**Pull request by soumesh113**
Close #21903

### CODE REVIEW

**Comment by soumesh113** [comment]
Hi @ZoeCD , could you please review this PR and suggest any changes to be made?

File: ivy/functional/frontends/tensorflow/math.py [hunk 1]
```diff
@@ -549,6 +549,19 @@ def tanh(x, name=None):
 def rsqrt(x, name=None):
     return ivy.reciprocal(ivy.sqrt(x))
 
+@to_ivy_arrays_and_back
+def segment_sum(data, segment_ids, name= "segment_sum"):
+    data = ivy.array(data)
```
Comment (ZoeCD) [inline ivy/functional/frontends/tensorflow/math.py:554]: You don't need to change the arrays to ivy arrays. The `to_ivy_arrays_and_back` decorator handles that

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py [hunk 2]
```diff
@@ -1940,6 +1940,35 @@ def test_tensorflow_rsqrt(
...
+    data,
+    segment_ids,
+    frontend,
+    test_flags,
+    fn_tree,
+    backend_fw,
+    on_device,
+):
+    helpers.test_frontend_function(
+        input_dtypes=["int32", "int64"],
```
Comment (ZoeCD) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py:1963]: Here you should be passing the dtype that your data has. I recommend checking other tests to see how it is done

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py [hunk 3]
```diff
@@ -1940,6 +1940,35 @@ def test_tensorflow_rsqrt(
...
+#segment_sum
+@handle_frontend_test(
+    fn_tree="tensorflow.math.segment_sum",
+    data=helpers.array_values(dtype=ivy.int32, shape=(5, 6), min_value=1, max_value=9),
+    segment_ids=helpers.array_values(
+        dtype=ivy.int32, shape=(5,), min_value=0, max_value=4
+    ),
+    test_with_out=st.just(False),
+)
+def test_tensorflow_segment_sum(
```
Comment (ZoeCD) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py:1952]: Hello! The test is failing for all backends. You can see the stacktrace and error in the details section of the intelligent-tests (1) 

Let me know if you have any questions!

**Review by ZoeCD — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py [hunk 4]
```diff
@@ -1940,6 +1940,43 @@ def test_tensorflow_rsqrt(
...
+    dtype_and_segment,
+    frontend,
+    test_flags,
+    fn_tree,
+    backend_fw,
+    on_device,
+):
+    data_dtype, data = dtype_and_data
+    segment_dtype, segment_ids = dtype_and_segment
+    helpers.test_frontend_function(
```
Comment (ZoeCD) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py:1970]: The tests are still failing. It shows` TypeError: 'Array' object cannot be interpreted as an integer`
You don't need to send the dtyès as an array. Check out this call as an example

``` input_dtype, x = dtype_and_x
      helpers.test_frontend_function(
        input_dtypes=input_dtype,
        backend_to_test=backend_fw,
        frontend=frontend,
        test_flags=test_flags,
        fn_tree=fn_tree,
        on_device=on_device,
        x1=x[0],
        x2=x[1],
    )

```
To the kwarg `input_dtypes` you only need to pass the `data_dtype`. And you pass the inputs at the end of the function call. And so on. Let me know if this makes sense!

**Review by ZoeCD — CHANGES_REQUESTED** [review]

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Pull request closed without merging**